### PR TITLE
Login signup user

### LINKED ISSUE #25: [2 pt] Create user login/signup API entrypoint
- Create `user_controller` and its route to handle front-end login/sign_up requests.
  - If the user exists, should return the user's data.
  - If doesn't exist, should create a new user and return the new user's data.
- Modify the `reservations#index` route and controller to display only reservations per user.
- Add uniqueness validation for `user_name`:
  - Not allowed two users with exactly the same name.
  - Case sensitive is allowed: `Peter != peter != PETER != PeTeR // => true`
- Update API documentation.

### PR DESCRIPTION
**Pull request by Camilovelag**
PR to close #25 💯 

### Future improvements / TBD ☑️ 
- Improve/customize the controller's json responses according to front-end future needs.
- Update API documentation (specs, yaml files).

### _Notes_
- Uniqueness validator implementation wasn't required indeed.

Happy coding 👋

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# frozen_string_literal: true
-
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
-      resources :reservations, only: [:index, :new, :create] do
+      resources :users, only: [:create] do
+        resources :reservations, only: [:index, :new, :create] do
```
Comment (JustoRago) [inline config/routes.rb:5]: The new path in reservations is not actually needed, there's no method by that name in the controller.

**Review by JustoRago — APPROVED** [review]
Excellent work @Camilovelag! We should merge this!

I have one small comment, see below.

**Review by cvalencia1991 — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Camilovelag   👋, 

Your project is complete!  There is nothing else to say other than... it's time to close it :shipit: 
Congratulations! 🎉 

leonardo-dicaprio

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: config/routes.rb (hunk 1 above)
Comment (Camilovelag) [reply config/routes.rb:5]: That's true, I'll delete this on `dev` branch.

**Pull request merged**